apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2.IntegrationTests/InsertTest.cs

```diff
@@ -491,5 +491,36 @@ public void CreateLoadJob_Parquet()
             var rows = table.ListRows().ToList();
             Assert.Equal(50, rows.Count);
         }
+
+        [Fact]
+        public void CreateLoadJobWithSession()
+        {
+            string sourceUri = "gs://cloud-samples-data/bigquery/us-states/us-states.parquet";
+
+            var client = BigQueryClient.Create(_fixture.ProjectId);
+            var tableRef = client.GetTableReference("_SESSION", "TempTestTable");
+
+            var options = new CreateLoadJobOptions { SourceFormat = FileFormat.Parquet, CreateSession = true };
+            var job = client.CreateLoadJob(sourceUri, tableRef, schema: null, options: options);
+            job.PollUntilCompleted().ThrowOnAnyError();
+
+            var sessionId = job.Statistics.SessionInfo.SessionId;
+            Assert.NotNull(sessionId);
+
+            var optionsWithConnectionProp = new CreateLoadJobOptions { SourceFormat = FileFormat.Parquet, SessionID = sessionId };
+            var jobWithConnectionProp = client.CreateLoadJob(sourceUri, tableRef, schema: null, options: optionsWithConnectionProp);
+            jobWithConnectionProp.PollUntilCompleted().ThrowOnAnyError();
+
+            Assert.NotNull(jobWithConnectionProp.Statistics.SessionInfo.SessionId);
+            Assert.Equal(sessionId, jobWithConnectionProp.Statistics.SessionInfo.SessionId);
+
+            // Checking if the data is loaded to the temp table in the session.
+            string queryTempTable = $"SELECT * FROM _SESSION.TempTestTable; ";
```

Comment:
Nit: I suspect we don't need the space at the end - do we even need the semi-colon after TempTestTable?

Reply:
done